mock/tests/e2e/App.spec.ts

```diff
@@ -0,0 +1,424 @@
+import { expect, test } from "@playwright/test";
+
+/**
+  The general shapes of tests in Playwright Test are:
+    1. Navigate to a URL
+    2. Interact with the page
+    3. Assert something about the page against your expectations
+  Look for this pattern in the tests below!
+ */
+
+// If you needed to do something before every test case...
+test.beforeEach(async ({ page }) => {
+  await page.goto("http://localhost:8000/");
```

Comment:
This test file is quite comphrehensive with consideration of lots of different input (both valid and invalid) combination. I just noticed a few of repetition when it comes to command testing, so I feel like it might be a better approach to devide the element checking into another test file and only use this test file for testing command-related things. In that way, the beforeEach can be more useful and reduce the repetitive operations